File: check_props.py
# -*- coding: utf-8 -*-
"""
Check actual database column types
"""
import requests, json, sys
sys.stdout.reconfigure(encoding='utf-8')

NOTION_KEY = "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"
h = {'Authorization': 'Bearer ' + NOTION_KEY, 'Notion-Version': '2025-09-03'}
DB_ID = "7ee379c2f4364d558b6b5b8c48d1b00b"

# Query an existing entry to see what properties exist
ds_id = "a5575563-f29c-4224-b6e8-1b804bf04ba6"
r = requests.post(f'https://api.notion.com/v1/data_sources/{ds_id}/query', headers=h, json={})
items = r.json().get('results', [])

print(f"Total entries: {len(items)}")

if items:
    entry = items[0]
    props = entry.get('properties', {})
    print(f"\nProperties from first entry:")
    for k, v in props.items():
        print(f"  {k}: type={v.get('type')}")

File: archive/check_props.js
// Check new database properties
const https = require('https');
const key = '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';
const dbId = '35491ad7-17ba-81cc-aa04-ce53f7234e17';

const req = https.request({
    hostname: 'api.notion.com',
    path: '/v1/databases/' + dbId,
    method: 'GET',
    headers: {
        'Authorization': 'Bearer ' + key,
        'Notion-Version': '2025-09-03'
    }
}, res => {
    let d = '';
    res.on('data', c => d += c);
    res.on('end', () => {
        const r = JSON.parse(d);
        if (r.properties) {
            Object.entries(r.properties).forEach(([k, v]) => {
                console.log('  ' + k + ' (' + v.type + ')');
            });
        } else {
            console.log(JSON.stringify(r));
        }
    });
});
req.end();
